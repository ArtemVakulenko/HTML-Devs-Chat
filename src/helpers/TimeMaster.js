export const getTimeString = (date) => {
  const msgDate = new Date(date);
  let month = msgDate.getMonth() + 1;
  let day = msgDate.getDate();
  let hours = msgDate.getHours();
  let minutes = msgDate.getMinutes();
  if (month < 10) month = `0${month}`;
  if (day < 10) day = `0${day}`;
  if (hours < 10) hours = `0${hours}`;
  if (minutes < 10) minutes = `0${minutes}`;
  const dateToShow = `${day}.${month} ${hours}:${minutes}`;
  return dateToShow;
};
