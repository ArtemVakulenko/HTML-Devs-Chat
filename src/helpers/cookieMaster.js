import { cookieAge } from '../constants';

export const setTokenInCookie = (payload, age = cookieAge) => {
    document.cookie = `token=${payload}; path=/; max-age=${age}`;
};
export const getCookie = (name) => {
    const value = `; ${document.cookie}`;
    const parts = value.split(`; ${name}=`);
    if (parts.length === 2) return parts.pop().split(';').shift();
};

export const deleteCookie = (name) => {
    setTokenInCookie(name, -1);
};
