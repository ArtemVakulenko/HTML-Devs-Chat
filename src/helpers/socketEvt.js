export const CREATE_ROOM = 'CREATE_ROOM';
export const NEW_MESSAGE = 'NEW_MESSAGE';
export const NEW_UNREAD_MESSAGE = 'NEW_UNREAD_MESSAGE';
