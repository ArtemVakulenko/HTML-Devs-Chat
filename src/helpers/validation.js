import { MESSAGES_KEYS, LOGIN, PASSWORD } from '../constants';

export const regValidate = ({ login, password, confirm }) => {
  if (!LOGIN.test(login)) {
    return MESSAGES_KEYS.invalidLogin;
  }
  if (!PASSWORD.test(password)) {
    return MESSAGES_KEYS.invalidPassword;
  }
  if (password !== confirm) {
    return MESSAGES_KEYS.invalidConfirm;
  }
  return false;
};

export const loginValidate = ({ login, password }) => {
  if (!LOGIN.test(login)) {
    return MESSAGES_KEYS.invalidLogin;
  }
  if (!PASSWORD.test(password)) {
    return MESSAGES_KEYS.invalidPassword;
  }
  return false;
};
