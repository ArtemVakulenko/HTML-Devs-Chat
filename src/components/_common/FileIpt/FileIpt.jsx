import React from 'react';
import PropTypes from 'prop-types';
import { useTranslation } from 'react-i18next';
import { useTheme } from 'src/components/hooks';
import { toBase64 } from 'helpers';
import { StFileIpt } from '../../Rooms/CreateRoomModal/ModalIpts/styled';

const FileIpt = ({ value, onChange }) => {
  const { colors, theme } = useTheme();
  const { t } = useTranslation();
  const onFileChange = async (e) => {
    const file = await toBase64(e.target.files[0]);
    onChange(e.target.files[0].name, file);
  };
  return (
    <>
      <StFileIpt htmlFor="upload" colors={colors} theme={theme}>{value || t('chooseFile')}</StFileIpt>
      <input type="file" id="upload" hidden onChange={onFileChange} />
    </>
  );
};

export default FileIpt;

FileIpt.propTypes = {
  value: PropTypes.string,
  onChange: PropTypes.func.isRequired,
};
