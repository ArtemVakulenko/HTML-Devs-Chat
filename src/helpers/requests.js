import { getCookie } from './cookieMaster';

export const postRequest = (url, body, method = 'POST') => {
  return fetch(url, {
    body: JSON.stringify(body),
    method,
    headers: {
      'Content-Type': 'application/json',
      Authorization: getCookie('token'),
    },
  });
};

export const getRequest = (url) => {
  return fetch(url, {
    headers: {
      Authorization: getCookie('token'),
    },
  });
};
