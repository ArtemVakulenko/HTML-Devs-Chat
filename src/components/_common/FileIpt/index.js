import FileIpt from './FileIpt';

export default FileIpt;
