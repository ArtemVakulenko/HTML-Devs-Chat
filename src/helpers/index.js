import { postRequest, getRequest } from './requests';
import { regValidate, loginValidate } from './validation';
import { setTokenInCookie, getCookie, deleteCookie } from './cookieMaster';
import { getTimeString } from './TimeMaster';
import * as EVT from './socketEvt';
import { toBase64 } from './toBase64';

export {
  postRequest, regValidate, loginValidate, setTokenInCookie,
  getCookie, getRequest, EVT, deleteCookie, toBase64, getTimeString,
};
