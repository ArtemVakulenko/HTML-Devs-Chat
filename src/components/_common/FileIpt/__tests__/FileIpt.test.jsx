import { shallow, mount } from 'enzyme';
import React from 'react';
import FileIpt from '../FileIpt';
import { toBase64 } from '../../../../helpers';

jest.mock('../../../../helpers', () => ({
  toBase64: jest.fn(),
}),
);
jest.mock('react-i18next', () => ({
  useTranslation: jest.fn().mockImplementation(() => ({
    t: jest.fn(),
  })),
}),
);
jest.mock('../../../hooks', () => ({
  useTheme: jest.fn().mockImplementation(() => ({
    colors: {
      dark: {
        color: '',
        input: '',
        placeholder: '',
      },
    },
    theme: 'dark',
  })),
}));
toBase64;
describe('FileIpt', () => {
  describe('type text', () => {
    const props = {
      value: 'MockValue',
      onChange: jest.fn(),
    };
    it('Should match snapshot', () => {
      const component = shallow(<FileIpt {...props} />);
      expect(component.html()).toMatchSnapshot();
    });
    it('Should have StFileIpt', () => {
      const component = mount(<FileIpt {...props} />);
      expect(component.find('styled__StFileIpt')).toHaveLength(1);
    });
    it('Should have input', () => {
      const component = mount(<FileIpt {...props} />);
      expect(component.find('input')).toHaveLength(1);
    });
  });
});
