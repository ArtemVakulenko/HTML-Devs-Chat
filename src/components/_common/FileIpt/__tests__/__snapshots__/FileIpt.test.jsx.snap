// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FileIpt type text Should match snapshot 1`] = `"<label for=\\"upload\\" class=\\"styled__StFileIpt-fLkDRX iPkIEE\\">MockValue</label><input type=\\"file\\" id=\\"upload\\" hidden=\\"\\"/>"`;
